src/wrapper.test.js

```diff
@@ -179,220 +189,328 @@ describe('Distro initialization', () => {
-              expect(resourceAttributeKeys).toContain('container.name');
-
-              expect(resource.attributes['aws.ecs.container.arn']).toBe(
-                'arn:aws:ecs:us-west-2:111122223333:container/0206b271-b33f-47ab-86c6-a0ba208a70a9'
-              );
-              expect(resource.attributes['aws.ecs.cluster.arn']).toBe(
-                'arn:aws:ecs:us-west-2:111122223333:cluster/default'
-              );
-              expect(resource.attributes['aws.ecs.launchtype']).toBe('EC2');
-              expect(resource.attributes['aws.ecs.task.arn']).toBe(
-                'arn:aws:ecs:us-west-2:111122223333:task/default/158d1c8083dd49d6b527399fd6414f5c'
-              );
-              expect(resource.attributes['aws.ecs.task.family']).toBe('curltest');
-              expect(resource.attributes['aws.ecs.task.revision']).toBe('26');
-            });
+          const { tracerProvider } = await wrapper.init;
+          const resource = tracerProvider.resource;
+
+          checkBasicResourceAttributes(resource);
+
+          const resourceAttributeKeys = Object.keys(resource.attributes);
+
+          // Default ECS from upstream detector
+          expect(resource.attributes['cloud.provider']).toBe('aws');
+          expect(resource.attributes['cloud.platform']).toBe('aws_ecs');
+
+          // These properties may left be blank in the test env
+          expect(resourceAttributeKeys).toContain('container.id');
+          expect(resourceAttributeKeys).toContain('container.name');
+
+          expect(resource.attributes['aws.ecs.container.arn']).toBe(
+            'arn:aws:ecs:us-west-2:111122223333:container/0206b271-b33f-47ab-86c6-a0ba208a70a9'
+          );
+          expect(resource.attributes['aws.ecs.cluster.arn']).toBe(
+            'arn:aws:ecs:us-west-2:111122223333:cluster/default'
+          );
+          expect(resource.attributes['aws.ecs.launchtype']).toBe('EC2');
+          expect(resource.attributes['aws.ecs.task.arn']).toBe(
+            'arn:aws:ecs:us-west-2:111122223333:task/default/158d1c8083dd49d6b527399fd6414f5c'
+          );
+          expect(resource.attributes['aws.ecs.task.family']).toBe('curltest');
+          expect(resource.attributes['aws.ecs.task.revision']).toBe('26');
         });
       });
     });
   });
 
   describe('On Amazon EKS', () => {
-    describe('on successful request', () => {
-      jest.mock('@opentelemetry/resource-detector-aws', () => {
-        return {
-          ...jest.requireActual('@opentelemetry/resource-detector-aws'), // import and retain the original functionalities
-          awsEksDetector: {
-            detect: jest.fn().mockReturnValue(mockedResource),
-          },
-        };
-      });
+    beforeEach(() => {
+      process.env.LUMIGO_REPORT_DEPENDENCIES = 'false';
+    });
 
+    describe('on successful request', () => {
       test('NodeTracerProvider should be given a resource with all the right attributes', async () => {
```

Comment:
test with the same name - 4

Reply:
It is not a problem, the hierarchy is different